fe-library-app/src/components/BookDetail/BookDetail.tsx

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { useNavigate, useParams } from 'react-router-dom'
+
+import CreateUpdateBookModal from '../../modals/CreateUpdateBookModal'
+import { BookByIdItemResponse, deleteBook, getBookById } from '../../services/BookService'
+import DefaultBookCover from '../BookList/BookCard/DefaultBookCover.png'
+import style from './BookDetail.module.css'
+
+const BookDetail = () => {
+  const { id } = useParams()
+  const [ book, setBook ] = useState<BookByIdItemResponse>()
+  const navigate = useNavigate()
+  const [ showUpdateBookModal, setShowUpdateBookModal ] = useState(false)
+
+  const getBookInfo = useCallback(() => {
```

Comment:
You can pass id as function prop and not pass it in dependencies array, that way you do not recreate function on id change

Reply:
Done.